src/components/views/settings/tabs/user/EncryptionUserSettingsTab.tsx

```diff
@@ -54,71 +61,115 @@ interface EncryptionUserSettingsTabProps {
     initialState?: State;
 }
 
+const useKeyBackupIsEnabled = (): boolean | undefined => {
+    const [isEnabled, setIsEnabled] = useState<boolean | undefined>(undefined);
+    const [loading, setLoading] = useState(true);
+
+    const matrixClient = useMatrixClientContext();
+
+    const checkStatus = useCallback(async () => {
+        const crypto = matrixClient.getCrypto()!;
+        const info = await crypto.getKeyBackupInfo();
+        setIsEnabled(Boolean(info?.version));
+    }, [matrixClient, setIsEnabled]);
+
+    useEffect(() => {
+        (async () => {
+            await checkStatus();
+            setLoading(false);
+        })();
+    }, [checkStatus, setLoading]);
+
+    useEventEmitter(matrixClient, ClientEvent.AccountData, (event: MatrixEvent): void => {
+        const type = event.getType();
+        // Recheck the status if this account data has been updated as this implies it has changed
+        if (type === "m.org.matrix.custom.backup_disabled") {
+            checkStatus();
+        }
+    });
+
+    return loading ? undefined : isEnabled;
+};
```

Comment:
I think we should merge the content of `useKeyBackupIsEnabled` and `useCheckEncryptionState` in one hook which can also host multiple custom hooks.

Also I think that `useKeyBackupIsEnabled` should use the `State` of the component, to have a better integration in the behaviour of the component.

Reply:
My usual take on this is that more, simpler functions are preferable to fewer larger ones as they're easier to understand as units of functionality, but I agree it is less code - I've changed it, see what you think.